.github/workflows/deploy.yml

```diff
@@ -81,73 +81,52 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
-        timeout-minutes: 1
-        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        uses: actions/checkout@v3
         with:
           ref: main
           fetch-depth: 0  # To fetch all commits
-      - name: Set up Java
-        uses: actions/setup-java@8df1039502a15bceb9433410b1a100fbe190c53b
         timeout-minutes: 1
+  
+      - name: Set up Java
+        uses: actions/setup-java@v3
         with:
           distribution: 'temurin'
           java-version: 17
+        timeout-minutes: 1
+  
       - name: Set up Gradle
-        uses: gradle/gradle-build-action@ac2d340dc04d9e1113182899e983b5400c17cda1
-        timeout-minutes: 10
+        uses: gradle/actions/setup-gradle@v2
         with:
-          gradle-home-cache-cleanup: true
+          gradle-version: 8.11
+          cache-gradle-home: true
+        timeout-minutes: 10
+  
       - name: Export Google Services JSON
         env:
           FIREBASE_DEBUG_JSON_BASE64: ${{ secrets.FIREBASE_DEBUG_JSON_BASE64 }}
           FIREBASE_RELEASE_JSON_BASE64: ${{ secrets.FIREBASE_RELEASE_JSON_BASE64 }}
-        if: "${{ env.FIREBASE_DEBUG_JSON_BASE64 != '' && env.FIREBASE_RELEASE_JSON_BASE64 != '' }}"
-        shell: bash
+        if: ${{ env.FIREBASE_DEBUG_JSON_BASE64 != '' && env.FIREBASE_RELEASE_JSON_BASE64 != '' }}
```

Comment:
If the syntax with double quotes causes a problem, note that it also occurs on lines 67-71.